pygeoapi/linked_data.py

```diff
@@ -173,104 +175,146 @@ def jsonldify_collection(cls, collection, locale_):
+        read from, and extended by, the pygeoapi configuration for a particular
+        dataset.
 
-    :param config: dict of configuration
-    :param data: dict of data:
-    :param dataset: dataset identifier
-    :param identifier: item identifier (optional)
-    :param id_field: item identifier_field (optional)
+        :param config: dict of configuration
+        :param data: dict of data:
+        :param dataset: dataset identifier
+        :param identifier: item identifier (optional)
+        :param id_field: item identifier_field (optional)
 
-    :returns: string of rendered JSON (GeoJSON-LD)
+        :returns: string of rendered JSON (GeoJSON-LD)
     """
-    context = config['resources'][dataset].get('context', [])
-    geojsonld = config['resources'][dataset].get('geojsonld', True)
+    context = config['resources'][dataset].get('context', []).copy()
+    defaultVocabulary = {
+        'schema': 'https://schema.org/',
+        id_field: '@id',
+        'type': '@type'
+    }
 
     if identifier:
         # Single geojsonld
-        if not geojsonld:
-            data, geocontext = make_jsonld(data)
-            data[id_field] = identifier
-        else:
-            data['id'] = identifier
+        defaultVocabulary.update({
+            'geosparql': 'http://www.opengis.net/ont/geosparql#'
+        })
+
+        # Expand properties block
+        data.update(data.pop('properties'))
+
+        # Include multiple geometry encodings
+        data['type'] = 'schema:Place'
+        jsonldify_geometry(data)
+        data[id_field] = identifier
 
     else:
         # Collection of geojsonld
+        defaultVocabulary.update({
+                'features': 'schema:itemListElement',
+                'FeatureCollection': 'schema:itemList'
+        })
+
         data['@id'] = '{}/collections/{}/items/'.format(
-            config['server']['url'], dataset)
+            config['server']['url'], dataset
+        )
+
         for i, feature in enumerate(data['features']):
+            # Get URI for each feature
             identifier = feature.get(id_field,
                                      feature['properties'].get(id_field, ''))
             if not is_url(str(identifier)):
                 identifier = '{}/collections/{}/items/{}'.format(
                     config['server']['url'], dataset, feature['id'])
 
-            if not geojsonld:
-                feature, geocontext = make_jsonld(feature)
-                geocontext.append({
-                    "features": "schema:itemListElement",
-                    "FeatureCollection": "schema:itemList"
-                })
-                # Note: @id or https://schema.org/url, both or something else?
-                feature[id_field] = identifier
-            else:
-                feature['id'] = identifier
-
-            data['features'][i] = feature
+            # Note: @id or https://schema.org/url, both or something else?
```

Comment:
Has this note been resolved now? (If so, remove.)

Reply:
we're going with `@id`

Reply:
amended